exercises/space-age/.meta/hints.md

```diff
@@ -4,3 +4,13 @@ Some Rust topics you may want to read about while solving this problem:
 
 - Traits, both the From trait and implementing your own traits
 - Default method implementations for traits
+- Macros, the use of a macro could reduce boilerplate and increase readability 
+  for this exercise. For instance, 
+  [a macro can implement a trait for multiple types at once](https://stackoverflow.com/questions/39150216/implementing-a-trait-for-multiple-types-at-once),
+  though it is fine to implement `years_during` in the Planet trait itself. A macro could
+  define both the structs and their implementations. Info to get started with macros can 
+  be found at:
+  [The Macros chapter in The Rust Programming Language](https://doc.rust-lang.org/stable/book/ch19-06-macros.html)
+  [an older version of the Macros chapter with helpful detail](https://doc.rust-lang.org/1.30.0/book/first-edition/macros.html).
+  [Rust By Example](https://doc.rust-lang.org/stable/rust-by-example/macros.html)
```

Comment:
Adding a suggestion for formatting, here. If you look at the [rendered hints](https://github.com/exercism/rust/blob/b038a0db9bd295bbd35cd865340065a48c23d68f/exercises/space-age/.meta/hints.md), you'll see that Markdown has stripped the whitespace. This should just improve the rendering.

```suggestion
  be found at:
  
  - [The Macros chapter in The Rust Programming Language](https://doc.rust-lang.org/stable/book/ch19-06-macros.html)
  - [an older version of the Macros chapter with helpful detail](https://doc.rust-lang.org/1.30.0/book/first-edition/macros.html).
  - [Rust By Example](https://doc.rust-lang.org/stable/rust-by-example/macros.html)
```

If you want, you can copy the change manually. Otherwise, you could commit this suggestion. Then, from your local machine, do 

```sh
git pull                                      # fetches the change from github back down to your branch
bin\configlet.exe generate . -o space-age     # regenerate the README again
git commit -am "regenerate the readme again"  # commit the change
git push                                      # push the changes back to github
```

Strongly recommend getting familiar with the git CLI. It's pretty abstruse to get started with, but once it clicks, it's really powerful.

Reply:
Thanks. Went through the steps and my fork and local did not change to the suggested commit.

```
C:\rust_projects\Exercism\rust>git pull
Merge made by the 'recursive' strategy.
 exercises/macros/tests/invalid/Cargo.toml          | 4 ++++
 exercises/macros/tests/invalid/comma-sep.rs        | 3 ++-
 exercises/macros/tests/invalid/double-commas.rs    | 3 ++-
 exercises/macros/tests/invalid/leading-comma.rs    | 3 ++-
 exercises/macros/tests/invalid/no-comma.rs         | 7 +++++++
 exercises/macros/tests/invalid/only-arrow.rs       | 3 ++-
 exercises/macros/tests/invalid/only-comma.rs       | 3 ++-
 exercises/macros/tests/invalid/single-argument.rs  | 3 ++-
 exercises/macros/tests/invalid/triple-arguments.rs | 1 +
 exercises/macros/tests/invalid/two-arrows.rs       | 1 +
 exercises/macros/tests/macros.rs                   | 6 ++++++
 11 files changed, 31 insertions(+), 6 deletions(-)
 create mode 100644 exercises/macros/tests/invalid/no-comma.rs

C:\rust_projects\Exercism\rust>bin\configlet generate . -o space-age

C:\rust_projects\Exercism\rust>git commit -am "regenerate the readme again"
On branch master
Your branch is ahead of 'upstream/master' by 3 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean

C:\rust_projects\Exercism\rust>git push
remote: Permission to exercism/rust.git denied to bobahop.
fatal: unable to access 'https://github.com/exercism/rust.git/': The requested URL returned error: 403

C:\rust_projects\Exercism\rust>git push origin
To https://github.com/bobahop/rust.git
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'https://github.com/bobahop/rust.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

C:\rust_projects\Exercism\rust>git pull upstream
Already up to date.
```

My config

```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "upstream"]
	url = https://github.com/exercism/rust.git
	fetch = +refs/heads/*:refs/remotes/upstream/*
[branch "master"]
	remote = upstream
	merge = refs/heads/master
[remote "origin"]
	url = https://github.com/bobahop/rust.git
	fetch = +refs/heads/*:refs/remotes/origin/*

```

Reply:
Perhaps it's because the suggested commit caused a check not to pass?

Reply:
Ah, didn't realize that you had a two-origin setup going on. You would have needed to push to `origin` instead of `upstream`, but because you're working on `master` branch, you've configured it to use `upstream` as the default remote. 

I just made the change and pushed it myself; it was simpler.